### PR TITLE
Participants list

### PR DESCRIPTION
**Pull request by nschloe**
Add a working participants list.

Various small fixes all over the place.

### CODE REVIEW

File: src/js/directives/gravatarList.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      templateUrl: 'templates/directives/gravatarList.html',
+      link: function(scope, element, attrs) {
+        scope.$watch('discussion', function(discussion) {
+          // TODO $watch doesn't fire on reply update
+          if (discussion !== undefined) {
+
+            scope.participants = _.map(
+              scope.discussion.replies,
+              function(reply) {return reply.author;}
+            );
```
Comment (andrenarchy) [inline src/js/directives/gravatarList.js]: shorter: `scope.participants = _.pluck(scope.discussion.replies, 'author');`

File: src/js/directives/gravatarList.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            // prepend original annotation author
+            scope.participants.unshift(
+              scope.discussion.originalAnnotation.author
+            );
+
+            // make list unique
+            scope.participants = _.uniq(
+              scope.participants,
+              function(user) {return user._id;}
+            );
```
Comment (andrenarchy) [inline src/js/directives/gravatarList.js]: Same here: `scope.participants = _.uniq(scope.participants, '_id');`

File: src/js/directives/gravatarList.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  app.directive('gravatarList', function() {
+    return {
+      restrict: 'E',
+      scope: {
+        discussion: '=',
+      },
+      templateUrl: 'templates/directives/gravatarList.html',
+      link: function(scope, element, attrs) {
+        scope.$watch('discussion', function(discussion) {
+          // TODO $watch doesn't fire on reply update
```
Comment (andrenarchy) [inline src/js/directives/gravatarList.js]: You can set the `objectEquality` parameter of `$watch` to `true`, see the [docs](https://docs.angularjs.org/api/ng/type/$rootScope.Scope).

**Comment by andrenarchy** [comment]
I see a few `pull-right` classes applied in the PR. The parent element should get the class `clearfix` because `pull-right` uses the `float` property. Otherwise, elements below the floated elements may be rendered awkwardly. Sources: [bootstrap docs](http://getbootstrap.com/css/#helper-classes-clearfix) and [SO](http://stackoverflow.com/questions/9543541/what-does-the-clearfix-class-do-in-css).

**Comment by nschloe** [comment]
This pull request doesn't introduce any `pull-right`s? Feel free to add `clearfix` where deemed needed though.

**Comment by andrenarchy** [comment]
Oh right... They've been around before! :-) Let's take care of them separately. I'll open an issue for this.

**Pull request merged**